lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -333,4 +344,32 @@ sub _check_error {
     return 0;
 }
 
+sub _for_every_batch {
+    my ($c, $project, $sub) = @_;
+    my $helper = $c->obs_rsync;
+    my ($parent_project, $thisbatch) = $helper->split_project_batch($project);
+
+    # Three cases possible when $sub() must be called:
+    # for this single batch alone ($thisbatch != undef)
+    # for a regular project without batches (!@$batches)
+    # for every batch of a project with batches
+    if ($thisbatch) {
+        return &$sub($parent_project, $thisbatch);
+    }
+    else {
```

Comment:
I'd avoid the `else` because the `if` block returns anyways. The same counts for the `else` within the `else`. Code with less nesting is usually easier to read. And the if could be written as `return &$sub($parent_project, $thisbatch) if $thisbatch;`.